src/EditorFeatures/CSharpTest/CodeActions/ConvertConversionOperators/ConvertConversionOperatorsFromCastTests.cs

```diff
@@ -0,0 +1,174 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+using VerifyCS = Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions.CSharpCodeRefactoringVerifier<Microsoft.CodeAnalysis.CSharp.CodeRefactorings.ConvertConversionOperators.CSharpConvertConversionOperatorsFromCastRefactoringProvider>;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings.ConvertConversionOperators
+{
+    [Trait(Traits.Feature, Traits.Features.ConvertConversionOperators)]
+    public class ConvertConversionOperatorsFromCastTests
+    {
+        [Fact]
+        public async Task ConvertFromExplicitToAs()
+        {
+            const string InitialMarkup = @"
+class Program
+{
+    public static void Main()
+    {
+        var x = ([||]object)1;
+    }
+}";
+            const string ExpectedMarkup = @"
+class Program
+{
+    public static void Main()
+    {
+        var x = 1 as object;
+    }
+}";
+            await new VerifyCS.Test
+            {
+                TestCode = InitialMarkup,
+                FixedCode = ExpectedMarkup,
+                CodeActionValidationMode = CodeActionValidationMode.Full,
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task ConvertFromExplicitToAs_ValueType()
+        {
+            const string InitialMarkup = @"
+class Program
+{
+    public static void Main()
+    {
+        var x = ([||]byte)1;
+    }
+}";
+            await new VerifyCS.Test
+            {
+                TestCode = InitialMarkup,
+                FixedCode = InitialMarkup,
+                OffersEmptyRefactoring = false,
+                CodeActionValidationMode = CodeActionValidationMode.None,
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task ConvertFromExplicitToAs_ValueTypeConstraint()
+        {
+            const string InitialMarkup = @"
+public class C
+{
+    public void M<T>() where T: struct
+    {
+        var o = new object();
+        var t = (T[||])o;
+    }
+}
+";
+            await new VerifyCS.Test
+            {
+                TestCode = InitialMarkup,
+                FixedCode = InitialMarkup,
+                OffersEmptyRefactoring = false,
```

Comment:
It is really hard to discover, that this means: "Run the refactoring provider and make sure no code actions are suggested". It is also easy to get wrong: e.g. if `FixedCode` is not set, the refactoring provider is not called at all.

It probably makes sense to add a method like `VerifyCS.VerifyMissingAsync(string markup)`. This would also help readability in the `ConvertIfToSwitch` tests:
https://github.com/dotnet/roslyn/blob/a34f425d15b9acd35e749d99b31c682fbb8ddf4b/src/EditorFeatures/CSharpTest/CodeActions/ConvertIfToSwitch/ConvertIfToSwitchTests.cs#L404-L418

`VerifyCS.VerifyRefactoringAsync(source, source)` does basically the same as the annotated code above, but without explicit setting `OffersEmptyRefactoring` to false.

Reply:
tagging @sharwell for feedback on the sdk test harness.